Sources/swift-algorand-sdk/v2/client/algod/GetGenesis.swift

```diff
@@ -5,66 +5,33 @@
 //  Created by Jesulonimi on 5/3/21.
 //
 
-
 import Foundation
 import Alamofire
 
-public class GetGenesis  {
-    var client:AlgodClient
-    init(client:AlgodClient) {
-        self.client=client
+public struct GetGenesis: Request {
+    public typealias ResponseType = String
+    public let client: HTTPClient
+    public let parameters: RequestParameters
+    init(client: AlgodClient) {
+        self.client = client
+        parameters = .init(path: "/genesis")
     }
 //
-    public func execute( callback: @escaping (_:Response<String>)->Void) {
-        
-    
-        let headers:HTTPHeaders=[client.apiKey:client.token]
-        
-        var request=AF.request(getRequestString(), method: .get, headers: headers,requestModifier: { $0.timeoutInterval = 120 })
-        var customResponse:Response<String>=Response()
-//      request.responseDecodable(of: String.self){ (response) in
-//        if(response.error != nil){
-//            customResponse.setIsSuccessful(value:false)
-//            var errorDescription=String(data:response.data ?? Data(response.error!.errorDescription!.utf8),encoding: .utf8)
-//            customResponse.setErrorDescription(errorDescription:errorDescription!)
-//            print(response.error)
-//            callback(customResponse)
-//            if(response.data != nil){
-//                if let message = String(data: response.data!,encoding: .utf8){
-//                    var errorDic = try! JSONSerialization.jsonObject(with: message.data, options: []) as? [String: Any]
-//
-//                    customResponse.errorMessage = errorDic!["message"] as! String
-//                }
-//            }
-//            return
-//        }
-//                        let data=response.value
-//
-//        customResponse.setData(data: (data ?? "nil") ?? "nil")
-//                        customResponse.setIsSuccessful(value:true)
-//                        callback(customResponse)
-//
-//        }
-        request.responseJSON(){response in
-//            print(String(bytes: response.data!,encoding: .utf8))
-            switch(response.result){
-            case .success(_): customResponse.setData(data: String(bytes: response.data!,encoding: .utf8) ?? "nil")
-                customResponse.setIsSuccessful(value:true)
+    public func execute( callback: @escaping (_:Response<String>) -> Void) {
+        let request = parameters.request(from: client)
+        let customResponse: Response<String> = Response()
+        request.responseData { response in
+            switch response.result {
```

Comment:
Is it because of the `responseData` that this class does not use the method in the Request extension?

Reply:
yeah exactly. It was the only method that had a different internal implementation before my changes.

Reply:
and the responseJson is deprecated and will be gone with Alamofire 6.0.

In my opinion we should eventually remove direct dependency to alamofire and use URLSession directly. But one thing at a time :D